sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -211,6 +211,50 @@ object GpuOrcScan extends Arm {
         } else {
           downCastAnyInteger(col, toDt)
         }
+
+      // bool to float, double(float64)
+      case (DType.BOOL8, DType.FLOAT32 | DType.FLOAT64) =>
+        col.castTo(toDt)
+
+      // bool to string
+      case (DType.BOOL8, DType.STRING) =>
+        withResource(col.castTo(toDt)) { casted =>
+          // cuDF produces "ture"/"false" while CPU outputs "TRUE"/"FALSE".
+          casted.upper()
+        }
+
+      // integer to float, double(float64), string
+      case (DType.INT8 | DType.INT16 | DType.INT32 | DType.INT64,
+      DType.FLOAT32 | DType.FLOAT64 | DType.STRING) =>
+        col.castTo(toDt)
+
+      // {bool, integer} to timestamp(micro seconds)
+      case (DType.BOOL8 | DType.INT8 | DType.INT16 | DType.INT32, DType.TIMESTAMP_MICROSECONDS) =>
+        // cuDF requires casting to a long first.
+        withResource(col.castTo(DType.INT64)) { longs =>
+          // In CPU, ORC assumes the integer value is in seconds, and returns timestamp in
+          // micro seconds.
+          withResource(longs.bitCastTo(DType.TIMESTAMP_SECONDS)) { seconds =>
+            seconds.castTo(DType.TIMESTAMP_MICROSECONDS)
+          }
+        }
+
+      // int64 to timestamp
+      case (DType.INT64, DType.TIMESTAMP_MICROSECONDS) =>
+        // We need overflow checking here, since max value of INT64 is about 9 * 1e18, and convert
+        // INT64 to micro seconds(also a INT64 actually), we need multiply 1e6, it may cause long
+        // integer-overflow.
+        // In Math.multiplyExact, if there is an integer-overflow, then it will throw an
+        // ArithmeticException.
+        // If these two 'Math.multiplyExact' throw no exception, it means no integer-overflow when
+        // casting 'col' to TIMESTAMP_MICROSECONDS. We did the similar thing as CPU code here.
+        val k = 1e6.toLong
+        Math.multiplyExact(col.max().getLong, k)
+        Math.multiplyExact(col.min().getLong, k)
+        withResource(col.bitCastTo(DType.TIMESTAMP_SECONDS)) { seconds =>
```

Comment:
I believe we can not simply convert integral to timestamp here.
Need to follow the casting from float to timesatmp in PR https://github.com/NVIDIA/spark-rapids/pull/6319, except the first section of float to integral.
I mean the casting should be like `integer -> milliseconds -> microseconds`, and check overflow only for the second conversion to follow what CPU does.

Reply:
Like this?
```scala
        withResource(Scalar.fromLong(1000)) { thousand =>
          withResource(col.mul(thousand)) { milliSeconds =>
            testLongOverflow(milliSeconds.max().getLong, 1000L)
            testLongOverflow(milliSeconds.min().getLong, 1000L)
            withResource(milliSeconds.mul(thousand)) { microSeconds =>
              microSeconds.castTo(DType.TIMESTAMP_MICROSECONDS)
            }
          }
        }
```